stories/core/Table.stories.tsx

```diff
@@ -304,6 +307,159 @@ Sortable.args = {
   isSortable: true,
 };
 
+export const LazyLoading: Story<TableProps> = (args) => {
+  const { columns, ...rest } = args;
+
+  const onClickHandler = (
+    props: CellProps<{ name: string; description: string }>,
+  ) => action(props.row.original.name)();
+
+  const tableColumns = useMemo(
+    () => [
+      {
+        Header: 'Table',
+        columns: [
+          {
+            id: 'name',
+            Header: 'Name',
+            accessor: 'name',
+          },
+          {
+            id: 'description',
+            Header: 'Description',
+            accessor: 'description',
+            maxWidth: 200,
+          },
+          {
+            id: 'click-me',
+            Header: 'Click',
+            width: 100,
+            Cell: (props: CellProps<{ name: string; description: string }>) => {
+              const onClick = () => onClickHandler(props);
+              return (
+                <a className='iui-anchor' onClick={onClick}>
+                  Click me!
+                </a>
+              );
+            },
+          },
+        ],
+      },
+    ],
+    [],
+  );
+
+  const generateData = (start: number, end: number) => {
+    return [...new Array(end - start)].map((_, index) => ({
+      name: `Name${start + index}`,
+      description: `Description${start + index}`,
+    }));
+  };
+
+  const [tableData, setTableData] = useState(() => generateData(1, 100));
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onBottomReached = useCallback(() => {
+    action('Bottom reached!')();
+    setIsLoading(true);
+    // Simulating request
+    setTimeout(() => {
+      setTableData(() => [
+        ...tableData,
+        ...generateData(tableData.length, tableData.length + 100),
+      ]);
+      setIsLoading(false);
+    }, 1000);
+  }, [tableData]);
+
+  return (
+    <Table
+      columns={columns || tableColumns}
+      emptyTableContent='No data.'
+      onBottomReached={onBottomReached}
+      isLoading={isLoading}
+      {...rest}
+      data={tableData}
+    />
+  );
+};
+
+LazyLoading.argTypes = {
+  data: { control: { disable: true } },
+  isLoading: { control: { disable: true } },
+};
+
+export const RowInView: Story<TableProps> = (args) => {
```

Comment:
I would suggest adding some description about this demo, if you want to advertise about intersection hook, cause this is demoing exactly it. You could say to pay attention in actions column to see, how intersection observer reports every row, when it is in viewport for the first time.